From c317d43586c49cf0ff9fcf8f3ad2235b616468e7 Mon Sep 17 00:00:00 2001
From: Peter Chiu <chui-hao.chiu@mediatek.com>
Date: Thu, 9 Jan 2025 09:29:54 +0800
Subject: [PATCH] wifi: mt76: mt7915: add sanity check to prevent invalid wlan
 id in txfree

Invalid wlan id would lead to kernel crash.
To prevent this issue, stop parsing txfree if wlan id is out of range.

Signed-off-by: Peter Chiu <chui-hao.chiu@mediatek.com>
---
 mt7915/mac.c | 3 +++
 1 file changed, 3 insertions(+)

--- a/mt7915/mac.c
+++ b/mt7915/mac.c
@@ -947,6 +947,9 @@ mt7915_mac_tx_free(struct mt7915_dev *de
 			u16 idx;
 
 			idx = FIELD_GET(MT_TX_FREE_WLAN_ID, info);
+			if (!idx || idx >= mt7915_wtbl_size(dev))
+				return;
+
 			wcid = rcu_dereference(dev->mt76.wcid[idx]);
 			sta = wcid_to_sta(wcid);
 			if (!sta)
